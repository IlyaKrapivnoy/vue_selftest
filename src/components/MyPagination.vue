<template>
  <div class="flex justify-center my-10">
    <div
      v-for="pageNumber in totalPages"
      :key="pageNumber"
      class="page-item"
      :class="{ 'current-page': page === pageNumber }"
      @click="changePage(pageNumber)"
    >
      {{ pageNumber }}
    </div>
  </div>
</template>

<script>
export default {
  name: "my-pagination",
  props: {
    page: {
      type: Number,
      required: true,
    },
    totalPages: {
      type: Number,
      required: true,
    },
  },
  methods: {
    changePage(pageNumber) {
      this.$emit("page-change", pageNumber);
    },
  },
};
</script>

<style>
.page-item {
  border: 1px solid black;
  padding: 2px 8px;
  border-radius: 5px;
  cursor: pointer;
  margin: 0 3px;
}

.page-item.current-page {
  border: 2px solid teal;
}
</style>
