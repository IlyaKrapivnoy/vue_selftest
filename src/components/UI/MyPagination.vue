<template>
  <div class="flex justify-center my-10">
    <div
      v-for="pageNumber in totalPages"
      :key="pageNumber"
      class="border border-black px-4 py-2 rounded-lg cursor-pointer mx-1"
      :class="{ 'current-page': page === pageNumber }"
      @click="changePage(pageNumber)"
    >
      {{ pageNumber }}
    </div>
  </div>
</template>

<script>
export default {
  name: "my-pagination",
  props: {
    page: {
      type: Number,
      required: true,
    },
    totalPages: {
      type: Number,
      required: true,
    },
  },
  methods: {
    changePage(pageNumber) {
      this.$emit("page-change", pageNumber);
    },
  },
};
</script>

<style>
.current-page {
  border: 2px solid teal;
}
</style>
