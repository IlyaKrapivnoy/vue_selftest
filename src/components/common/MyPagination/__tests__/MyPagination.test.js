import { mount } from "@vue/test-utils";
import MyPagination from "@/components/common/MyPagination/MyPagination.vue";

describe("MyPagination", () => {
  it("matches snapshot", () => {
    const wrapper = mount(MyPagination, {
      props: {
        page: 1,
        totalPages: 5,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("emits page-change event when a page is clicked", async () => {
    const wrapper = mount(MyPagination, {
      props: {
        page: 3,
        totalPages: 7,
      },
    });

    const pageItem = wrapper.find(".page-item:nth-child(4)");
    await pageItem.trigger("click");

    expect(wrapper.emitted("page-change")).toBeTruthy();
    expect(wrapper.emitted("page-change")[0][0]).toBe(4);
  });
});
