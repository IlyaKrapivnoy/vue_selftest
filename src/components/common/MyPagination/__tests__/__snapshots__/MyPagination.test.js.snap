// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPagination matches snapshot 1`] = `
"<div class="flex justify-center my-10">
  <div class="page-item current-page">1</div>
  <div class="page-item">2</div>
  <div class="page-item">3</div>
  <div class="page-item">4</div>
  <div class="page-item">5</div>
</div>"
`;
